src/components/dialog/dialog.component.js

```diff
@@ -170,8 +170,8 @@ class Dialog extends Modal {
   }
 
   get closeIcon() {
-    const { showCloseIcon, onCancel } = this.props;
-    if (!showCloseIcon || !onCancel) return null;
+    const { showCloseIcon, onCancel, disableCancel } = this.props;
```

Comment:
You've just added an extra prop to `Dialog` here... and not documented it anywhere. 
Disabling the close icon should be handled in Confirm itself, not like this